lib/analytics/load_service.rb

```diff
@@ -0,0 +1,84 @@
+require 'google/apis/analytics_v3'
+require 'googleauth'
+require 'analytics/google_analytics_export_credentials'
+
+module Analytics
+  class LoadService
+    include Google::Apis::AnalyticsV3
+    include Google::Auth
+
+    SCOPES = ['https://www.googleapis.com/auth/analytics.edit'].freeze
+
+    attr_reader :service
+
+    def initialize
+      @service = AnalyticsService.new # Google::Apis::AnalyticsV3::AnalyticsService
+      @service.authorization = GoogleAnalyticsExportCredentials.authorization(SCOPES)
+    end
+
+    def upload_csv(csv_data)
+      assert_has_required_env_vars
+
+      response = ''
+      Tempfile.create(['search_api_indexed_pages', '.csv'], export_path) do |file|
+        file.write(csv_data)
+        response = service.upload_data(
```

Comment:
We call this method here https://github.com/googleapis/google-api-ruby-client/blob/0.30.8/generated/google/apis/analytics_v3/service.rb#L2991